scripts/protect_branch_git_hook.py

```diff
@@ -0,0 +1,31 @@
+import re
+import subprocess
+import sys
+from typing import NoReturn
+
+
+def ProtectBranches() -> NoReturn:
+    hookid = "protect-branches"
+    protected_branches = [r"main", r"branch-\d+\.\d+"]
+    current_branch = (
+        subprocess.run(["git", "branch", "--show-current"], capture_output=True)
+        .stdout.decode(sys.stdout.encoding)
+        .replace("\n", "")
+    )
```

Comment:
We have a long line limit, I think this can all just go on a single line, without any parens for line continuation (especially if you use `from subprocess import run` at the top).

Reply:
Changed it but to me it was clearer as it was before. Anyway I really think these kind of comments should disappear by using an autoformatter as proposed in https://github.com/bokeh/bokeh/issues/10712.